### DISCUSSION TITLE
Does fluent-package(5.1.0-1.el8) or td-agent (4.5.2-1.el8) x86_64 of RHEL8 (or any linux) installed openssl.so points to OS installed openssl version?

### DISCUSSION THREAD

**Discussion by deepaksb2** [fluent/fluentd#4656] [Q&A]
### What is a problem?

The scan vulnerability on openssl.so (/opt/..../../x86_64-linux/openssl.so) shows the version which is installed on the OS and reports CVE-2024-5535. 
My question is the openssl.so by default points to the Openssl version installed on linux OS?
What is the openssl version installed by the fluent package ? if yes then how to point to that openssl version? 

### Describe the configuration of Fluentd

> /opt/td-agent/lib/ruby/2.7.0/x86_64-linux/strings openssl.so | grep OpenSSL command shows below output which matches with OS openssl version 
OpenSSL_version
OpenSSL
OpenSSLError
OpenSSL 1.1.1k  FIPS 25 Mar 2021

 > openssl version -a
OpenSSL 1.1.1k  FIPS 25 Mar 2021
built on: Thu Nov 30 13:05:10 2023 UTC
platform: linux-x86_64

OS Release details:
NAME="Red Hat Enterprise Linux"
VERSION="8.10 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.10"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.10 (Ootpa)"
ANSI_COLOR="0;31"

### Environment

```markdown
- Fluentd version: fluent-package-5.1.0-1.el8.x86_64.rpm
- TD Agent version:td-agent-4.5.2-1.el8.x86_64.rpm
- Fluent Package version:
- Docker image (tag):
- Operating system: RHEL 8
- Kernel version:
```

**Comment by deepaksb2** [comment]
Please help me on this query.

**Comment by sam-hieken** [comment]
OP I'm not familiar with the inner workings of Fluent but this may help: https://www.fluentd.org/blog/upgrade-td-agent-v4-to-v5 "Differences between td-agent v4 and fluent-package v5"

**Reply by deepaksb2** [reply]
I installed fluent5 but still observe the same behavior.

**Comment by cosmo0920** [comment]
> My question is the openssl.so by default points to the Openssl version installed on linux OS?

Yes. fluent-package depends on OS provided OpenSSL. And openssl shared object for importing OpenSSL features into Ruby is also depending on OS's openssl library.
